sklearn/utils/_test_common/instance_generator.py

```diff
@@ -0,0 +1,438 @@
+    RFECV: dict(cv=3),
+    # be tolerant of noisy datasets (not actually speed)
+    SelectFdr: dict(alpha=0.5),
+    # SelectKBest has a default of k=10
+    # which is more feature than we have in most case.
+    SelectKBest: dict(k=1),
+    SelfTrainingClassifier: dict(max_iter=5),
+    SequentialFeatureSelector: dict(cv=3),
+    SGDClassifier: dict(max_iter=5),
+    SGDOneClassSVM: dict(max_iter=5),
+    SGDRegressor: dict(max_iter=5),
+    SparsePCA: dict(max_iter=5),
+    # Due to the jl lemma and often very few samples, the number
+    # of components of the random matrix projection will be probably
+    # greater than the number of features.
+    # So we impose a smaller number (avoid "auto" mode)
+    SparseRandomProjection: dict(n_components=2),
+    SpectralBiclustering: dict(n_init=2, n_best=1, n_clusters=2),
+    SpectralClustering: dict(n_init=2, n_clusters=2),
+    SpectralCoclustering: dict(n_init=2, n_clusters=2),
+    # Default "auto" parameter can lead to different ordering of eigenvalues on
+    # windows: #24105
+    SpectralEmbedding: dict(eigen_tol=1e-5),
+    StackingClassifier: dict(cv=3),
+    StackingRegressor: dict(cv=3),
+    SVC: dict(max_iter=-1),
+    SVR: dict(max_iter=-1),
+    TargetEncoder: dict(cv=3),
+    TheilSenRegressor: dict(max_iter=5, max_subpopulation=100),
+    # TruncatedSVD doesn't run with n_components = n_features
+    TruncatedSVD: dict(n_iter=5, n_components=1),
+    TSNE: dict(perplexity=2),
+    TunedThresholdClassifierCV: dict(cv=3),
+    TweedieRegressor: dict(max_iter=5),
+}
+
+
+def _set_checking_parameters(estimator):
+    # set parameters to speed up some estimators and
+    # avoid deprecated behaviour
+    if type(estimator) in TEST_PARAMS:
+        test_params = TEST_PARAMS[type(estimator)]
+        estimator.set_params(**test_params)
+
+
+def _tested_estimators(type_filter=None):
+    for name, Estimator in all_estimators(type_filter=type_filter):
+        try:
+            estimator = _construct_instance(Estimator)
+        except SkipTest:
+            continue
+
+        yield estimator
+
+
+def _generate_pipeline():
+    for final_estimator in [Ridge(), LogisticRegression()]:
+        yield Pipeline(
+            steps=[
+                ("scaler", StandardScaler()),
+                ("final_estimator", final_estimator),
+            ]
+        )
+
+
+def _construct_instance(Estimator):
+    """Construct Estimator instance if possible."""
+    required_parameters = getattr(Estimator, "_required_parameters", [])
+    if len(required_parameters):
+        if required_parameters in (["estimator"], ["base_estimator"]):
+            # `RANSACRegressor` will raise an error with any model other
+            # than `LinearRegression` if we don't fix `min_samples` parameter.
+            # For common test, we can enforce using `LinearRegression` that
+            # is the default estimator in `RANSACRegressor` instead of `Ridge`.
+            if issubclass(Estimator, RANSACRegressor):
```

Comment:
Not in this PR because it'll go sideways but I think that we should have a small subsequent PR where we should specify as much as possible the parameter below into the `TEST_PARAM` and make sure that we call the `_set_checking_parameters`. This would be more consistent with the rest and reduce the number of places that we set parameters.